src/test/polymer/polymer2-element-scanner_test.ts

```diff
@@ -14,70 +14,59 @@
 
 
 import {assert} from 'chai';
-import * as fs from 'fs';
 import * as path from 'path';
 
 import {Visitor} from '../../javascript/estree-visitor';
-import {JavaScriptDocument} from '../../javascript/javascript-document';
 import {JavaScriptParser} from '../../javascript/javascript-parser';
-import {ScannedElement, ScannedFeature} from '../../model/model';
+import {ScannedFeature} from '../../model/model';
+import {ScannedPolymerElement} from '../../polymer/polymer-element';
 import {Polymer2ElementScanner} from '../../polymer/polymer2-element-scanner';
-
-function compareTagNames(a: {tagName?: string}, b: {tagName?: string}): number {
-  const tagNameA = a && a.tagName;
-  const tagNameB = b && b.tagName;
-
-  if (tagNameA == null)
-    return (tagNameB == null) ? 0 : -1;
-  if (tagNameB == null)
-    return 1;
-  return tagNameA.localeCompare(tagNameB);
-};
+import {FSUrlLoader} from '../../url-loader/fs-url-loader';
+import {CodeUnderliner} from '../test-utils';
 
 suite('Polymer2ElementScanner', () => {
+  const testFilesDir = path.resolve(__dirname, '../static/polymer2/');
+  const urlLoader = new FSUrlLoader(testFilesDir);
+  const underliner = new CodeUnderliner(urlLoader);
 
-  let document: JavaScriptDocument;
-  let elements: Map<string|undefined, ScannedElement>;
-  let elementsList: ScannedElement[];
-
-  suiteSetup(async() => {
-    const parser = new JavaScriptParser();
-    const file = fs.readFileSync(
-        path.resolve(__dirname, '../static/polymer2/test-element.js'), 'utf8');
-    document = parser.parse(file, '/static/polymer2/test-element.js');
-    const scanner = new Polymer2ElementScanner();
-    const visit = (visitor: Visitor) =>
-        Promise.resolve(document.visit([visitor]));
+  async function getElements(filename: string):
+      Promise<ScannedPolymerElement[]> {
+        const file = await urlLoader.load(filename);
+        const parser = new JavaScriptParser();
+        const document = parser.parse(file, filename);
+        const scanner = new Polymer2ElementScanner();
+        const visit = (visitor: Visitor) =>
+            Promise.resolve(document.visit([visitor]));
 
-    const features: ScannedFeature[] = await scanner.scan(document, visit);
-    elements = new Map();
-    elementsList =
-        <ScannedElement[]>features.filter((e) => e instanceof ScannedElement);
-    for (const element of elementsList) {
-      elements.set(element.tagName, element);
-    }
-  });
+        const features: ScannedFeature[] = await scanner.scan(document, visit);
+        return <ScannedPolymerElement[]>features.filter(
```

Comment:
Google style is to use `as` for casting rather than `<>`. I recommend we converge